commons/all/src/main/java/org/infinispan/commons/jdkspecific/OffHeapMemory.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.commons.jdkspecific;
+
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.util.Util;
+
+/**
+ * Class used to retrieve an instance of {@link org.infinispan.commons.spi.OffHeapMemory} based on the current
+ * running version of the JVM.
+ */
+public class OffHeapMemory {
+   private static final org.infinispan.commons.spi.OffHeapMemory INSTANCE = createInstance();
+
+   public static org.infinispan.commons.spi.OffHeapMemory getInstance() {
+      return INSTANCE;
+   }
+
+   private static org.infinispan.commons.spi.OffHeapMemory createInstance() {
```

Comment:
I don't like this.
You can have this version of OffHeapMemory return the unsafe instance, and the one in the jdk21 jar return the FFM variant. No need for reflection: the class loader will give you the highest available version

Reply:
That is how I thought it would work as well, but I was just copying what we had with the ThreadCreator class, but I guess that one is different as it can be disabled.

Reply:
@tristantarrant changed it up, let me know what you think now.